libdnf5/transaction/transaction.cpp

```diff
@@ -170,13 +170,17 @@ void Transaction::fill_transaction_environments(
     for (auto & tsenv : transaction_environments) {
         auto & new_env = new_comps_environment();
         auto environment = tsenv.get_environment();
+        new_env.set_environment_id(environment.get_environmentid());
         new_env.set_name(environment.get_name());
         new_env.set_translated_name(environment.get_translated_name());
         libdnf5::comps::PackageType package_types{libdnf5::comps::PackageType::MANDATORY};
         if (tsenv.get_with_optional()) {
             package_types |= libdnf5::comps::PackageType::OPTIONAL;
         }
         new_env.set_package_types(package_types);
+        if (tsenv.get_environment().get_repos().begin() != tsenv.get_environment().get_repos().end()) {
```

Comment:
Looks good. I would use `tsenv.get_environment().get_repos().empty()` here and below.

Reply:
I like it, thanks!
Updated.